scalableminds/webknossos
# Followup fixes for "collapse/expand all subgroups"

This PR adds small fixes/changes to the behavior of the "collapse/expand all subgroups" feature. 

If you expand all subgroups, the group itself is now also automatically expanded. 
Previously it was possible to expand all subgroups of a collapsed group, but without expanding the group itself. This needed an extra "expand this group click".

Now the groups deeply check their children whether for expanded/collapsed groups. Previously only the immediate children were checked.


### URL of deployed dev instance (used for testing):
- [https://addcollapseallgroups.webknossos.xyz](https://addcollapseallgroups.webknossos.xyz)

### Steps to test:
- open a skeleton tracing
-  test the scenario shown in the picture of the issue #4166. In this specific scenario, it should be possible to still expand all children of the root group.
- use Expand all subgroups" on a collapsed group. The group itself should now also automatically expand itself.

### Issues:
- fixes #4166

------
